oneview/data_source_server_profile.go

```diff
@@ -1251,12 +1254,14 @@ func dataSourceServerProfileRead(d *schema.ResourceData, meta interface{}) error
 	d.Set("mac_type", serverProfile.MACType)
 
 	// Management Processor
-	mpSettings := make([]interface{}, 0, len(serverProfile.ManagementProcessor.MpSettings))
-	for _, mpSetting := range serverProfile.ManagementProcessor.MpSettings {
-		mpSettings = append(mpSettings, map[string]interface{}{
-			"args":         mpSetting.Args,
-			"setting_type": mpSetting.SettingType,
-		})
+	mpSettings := make([]interface{}, 0)
```

Comment:
any specific reason why have we removed  len(serverProfile.ManagementProcessor.MpSettings ?

Reply:
Giving the third argument "capacity" is redundant as we are anyway looping over the length of the object.
The next line is a loop
for _, mpSetting := range serverProfile.ManagementProcessor.MpSettings {